### PR TITLE
Nginx Server: Arrange the discovery backend datas and refactor Nginx configuration

### PR DESCRIPTION
**Pull request by xiekeyang**
This move all nginx server introduction to /contrib/nginx/README.md.

TODO: Should place
https://github.com/xiekeyang/oci-discovery/blame/master/README.md#L172-L205
also to contrib/nginx/README.md

Signed-off-by: xiekeyang <xiekeyang@huawei.com>

### CODE REVIEW

**Comment by wking** [comment]
I've pushed #51 with some suggested changes for this PR.  If they look good to you, squash them into your commit here.  There's no need to explicitly merge #51 (avoiding #17 and the resulting squash commit getting preserved in our master history).

**Comment by xiekeyang** [comment]
See, thanks much!

**Comment by xiekeyang** [comment]
Updated to https://github.com/xiekeyang/oci-discovery/pull/50/commits/5dbb551ca7154592039c6c5a4152fce3322741bf, https://github.com/xiekeyang/oci-discovery/pull/50/commits/275bdd1c5c4e6544276889ba04ab96a008eda3d7 and https://github.com/xiekeyang/oci-discovery/pull/50/commits/1eaa98c41fc0c319b906eff263085be9b76bbe2a.

**Comment by xiekeyang** [comment]
LGTM

File: contrib/nginx/README.md [hunk 1]
```diff
@@ -1,12 +1,56 @@
...
+
+With the [example ref-engines object](example.com/.well-known/oci-host-ref-engines), consumers will attempt to resolve image names matching the `example.com/app#…` family of [host-based image names](../../host-based-image-names.md) via an [OCI Index Template](../../index-template.md) ref engine at `https://example.com/oci-index/app`.
+That [index][] is served from `/srv/example.com/oci-index/app` with example content like [this](example.com/oci-index/app).
+
+The `org.opencontainers.image.ref.name` value in [the example index](example.com/oci-index/app) assumes consumers will only be attempting to match the `fragment` and not the full image name; image-spec does not currently provide guidance on this point.
+
+The `casEngines` entry in [the example index](example.com/oci-index/app) suggests consumers retrieve CAS blobs from `https://example.com/oci-cas/{algorithm}/{encoded:2}/{encoded}`.
+Supply those blobs at `/srv/example.com/oci-cas/{alg}/{prefix}/{encoded:2}/{encoded}`.
```
Comment (wking) [inline contrib/nginx/README.md]: Oops, this should have been `{algorithm}/{encoded:2}/{encoded}`, not `{alg}/{prefix}/{encoded:2}/{encoded}`.

File: contrib/nginx/README.md [hunk 2]
```diff
@@ -1,12 +1,56 @@
...
+For example `sha256:e9770a03fbdccdd4632895151a93f9af58bbe2c91fdfaaf73160648d250e6ec3` is at [`/srv/example.com/oci-cas/sha256/e9/e9770a03fbdccdd4632895151a93f9af58bbe2c91fdfaaf73160648d250e6ec3`](example.com/oci-cas/sha256/e9/e9770a03fbdccdd4632895151a93f9af58bbe2c91fdfaaf73160648d250e6ec3).
+It would be more conformant [if that content was canonical JSON][image-spec-canonical-json], but I've added newlines and indents to make the example more readable.
+
+To publish additional images matching the `example.co/app#…` family of [host-based image names](../../host-based-image-names.md), add their entries to [`/srv/example.com/oci-index/app`](example.com/oci-index/app)'s `manifests` array.
```
Comment (wking) [inline contrib/nginx/README.md]: `example.co` -> `example.com`

Comment (xiekeyang) [reply contrib/nginx/README.md]: Done by 5a6828b366ded8209eb56c7584c3ceae6fbe3ab4

File: contrib/nginx/README.md (hunk 1 above)
Comment (wking) [reply contrib/nginx/README.md]: Filed a fix in #52.

**Pull request merged**